File: spec/test_parsing_lists.c
#include "testparser.h"
#include <assert.h>
#include <string.h>

int main()
{
    char const * input = "a";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 1u);
    p_context_delete(context);

    input = "";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 0u);
    p_context_delete(context);

    input = "aaaaaaaaaaaaaaaa";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 16u);
    p_context_delete(context);

    return 0;
}

File: spec/test_parsing_lists.rs
use testparser::*;

fn main() {
    let cases: [(&[u8], u32); 3] = [(b"a", 1), (b"", 0), (b"aaaaaaaaaaaaaaaa", 16)];
    for (input, expected) in cases {
        let mut context = p_context_new(input);
        assert_eq!(P_SUCCESS, p_parse(&mut context));
        assert_eq!(expected, p_result(&context));
        p_context_delete(context);
    }
}
